src/consensus/produce_block.rs

```diff
@@ -0,0 +1,246 @@
+use std::time::{Duration, Instant};
+
+use color_eyre::Result;
+use fastrace::Span;
+use log::{info, warn};
+
+use crate::MAX_TRANSACTION_SIZE;
+use crate::crypto::Hash;
+use crate::network::NetworkMessage;
+use crate::shredder::{MAX_DATA_PER_SLICE, RegularShredder, Shredder, Slice};
+use crate::{All2All, Disseminator, Slot, network::Network};
+
+use super::{Alpenglow, DELTA_BLOCK};
+
+enum Continue {
+    Continue { left: Duration },
+    Stop,
+}
+
+async fn produce_slice<T>(
+    txs_receiver: &T,
```

Comment:
I feel like `produce_slice` should not be the one directly reading from the network, e.g. replacing this with a channel might be a way but we can also keep that for alter.

Reply:
Yeah, I think something like that makes sense and will help with testing too.  I think we can add something as a follow up.